scipy/stats/tests/test_stats.py

```diff
@@ -3723,123 +3677,158 @@ def do(self, a, b, axis=None, dtype=None):
         #  Test a 2d masked array
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = 52.8885199
-        self.do(np.ma.array(a), b)
+        self.equal_test(np.ma.array(a), b)
 
-    def test_2daxis0(self):
+    def test_2d_axis0(self):
         #  Test a 2d list with axis=0
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = np.array([35.56893304, 49.32424149, 61.3579244, 72.68482371])
-        self.do(a, b, axis=0)
+        self.equal_test(a, b, axis=0)
 
-    def test_2daxis1(self):
+        a = array([[1, 2, 3, 4], [1, 2, 3, 4], [1, 2, 3, 4]])
+        desired = array([1, 2, 3, 4])
+        self.equal_test(a, desired, axis=0, decimal=14)
+
+        a = array([[1, 2, 3, 4], [1, 2, 3, 4], [1, 2, 3, 4]])
+        desired = stats.gmean(a, axis=0)
+        self.equal_test(a, desired, axis=0, decimal=14)
+
+    def test_2d_axis1(self):
         #  Test a 2d list with axis=1
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = np.array([22.13363839, 64.02171746, 104.40086817])
-        self.do(a, b, axis=1)
+        self.equal_test(a, b, axis=1)
 
-    def test_2dmatrixdaxis0(self):
+        a = array([[1, 2, 3, 4], [1, 2, 3, 4], [1, 2, 3, 4]])
+        v = power(1 * 2 * 3 * 4, 1. / 4.)
+        desired = array([v, v, v])
+        self.equal_test(a, desired, axis=1, decimal=14)
+
+    def test_2d_matrix_axis0(self):
         #  Test a 2d list with axis=0
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = np.matrix([[35.56893304, 49.32424149, 61.3579244, 72.68482371]])
-        self.do(np.matrix(a), b, axis=0)
+        self.equal_test(np.matrix(a), b, axis=0)
 
-    def test_2dmatrixaxis1(self):
+    def test_2d_matrix_axis1(self):
         #  Test a 2d list with axis=1
         a = [[10, 20, 30, 40], [50, 60, 70, 80], [90, 100, 110, 120]]
         b = np.matrix([[22.13363839, 64.02171746, 104.40086817]]).T
-        self.do(np.matrix(a), b, axis=1)
+        self.equal_test(np.matrix(a), b, axis=1)
 
-    def test_1dlist0(self):
+    def test_large_values(self):
+        a = array([1e100, 1e200, 1e300])
+        self.approx_test(a, 1e200, significant=13)
+
+    def test_1d_list0(self):
         #  Test a 1d list with zero element
         a = [10, 20, 30, 40, 50, 60, 70, 80, 90, 0]
         b = 0.0  # due to exp(-inf)=0
         olderr = np.seterr(all='ignore')
         try:
-            self.do(a, b)
+            self.equal_test(a, b)
         finally:
             np.seterr(**olderr)
 
-    def test_1darray0(self):
+    def test_1d_array0(self):
         #  Test a 1d array with zero element
         a = np.array([10, 20, 30, 40, 50, 60, 70, 80, 90, 0])
         b = 0.0  # due to exp(-inf)=0
         olderr = np.seterr(all='ignore')
         try:
-            self.do(a, b)
+            self.equal_test(a, b)
         finally:
             np.seterr(**olderr)
 
-    def test_1dma0(self):
+    def test_1d_ma0(self):
```

Comment:
There are two test libraries ``test_stats`` for regular statistics and ``test_mstats_basic`` for masked arrays. The masked array test should be moved to ``test_mstats_basic``.

Reply:
OK. I will fix them.